packages/vite/src/node/plugins/css.ts

```diff
@@ -439,12 +439,61 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
   }
 }
 
+const createStyleContentMap = () => {
+  const contents = new Map<string, string>() // css id -> css content
+  const scopedIds = new Set<string>() // whether that id of css is scoped
+  const relations = new Map<
+    /* the id of the target for which css is scoped to */ string,
+    Array<{
+      /** css id */ id: string
+      /** export name */ exp: string | undefined
+    }>
+  >()
+
+  return {
+    putContent(
+      id: string,
+      content: string,
+      scopeTo: CustomPluginOptionsVite['cssScopeTo'] | undefined,
+    ) {
+      contents.set(id, content)
+      if (scopeTo) {
+        const [scopedId, exp] = scopeTo
+        if (!relations.has(scopedId)) {
+          relations.set(scopedId, [])
+        }
+        relations.get(scopedId)!.push({ id, exp })
+        scopedIds.add(id)
+      }
+    },
+    hasContentOfNonScoped(id: string) {
+      return !scopedIds.has(id) && contents.has(id)
+    },
+    getContentOfNonScoped(id: string) {
+      if (scopedIds.has(id)) return
+      return contents.get(id)
+    },
+    hasContentsScopedTo(id: string) {
+      return (relations.get(id) ?? [])?.length > 0
+    },
+    getContentsScopedTo(id: string) {
+      const rels = [...(relations.get(id) ?? [])]
+      // sort to get a deterministic output
+      rels.sort((a, b) => (a.id > b.id ? 1 : -1))
```

Comment:
I think this could potentially cause bugs where a component module imports multiple scoped css modules, as the import order should be respected in the component module 🤔 

Perhaps we can use `getModuleInfo` of `id` to try and determine the order of the `rels.id`?

I'm also fine with kicking this off to fixing later if it's tricky to solve now as long as we add a comment above.

Reply:
Oh, that's right. I wrongly assumed that the scoped styles does not interact each other.
`this.getModuleInfo(id).importedIds` seems to work.